### PR TITLE
Backport Kafka access log to v0.12

### PR DESCRIPTION
**Pull request by manalibhutiyani**
Issue #1937 Backport Kafka access log to v0.12
Fixes: #1937

```release-note
Kafka access log support
```

### CODE REVIEW

File: pkg/policy/api/selector.go
Comment (aanm) [inline pkg/policy/api/selector.go]: It's seems at least this commit was created for this branch, you should merge this commit first on master and then back port it to `v0.12`

**Review by aanm — CHANGES_REQUESTED** [review]

File: pkg/policy/api/selector.go
Comment (manalibhutiyani) [reply pkg/policy/api/selector.go]: I cannot merge this into master. The new logging changes done by Ray in master are not to be back-ported and this was a way to get around the merge conflicts that created.

Comment (aanm) [reply pkg/policy/api/selector.go]: My comment was for the whole commit, I was also referring on the method's descriptions changes bellow.

**Review by tgraf — APPROVED** [review]

**Comment by manalibhutiyani** [comment]
@raybejjani @jrajahalme I need an approval from proxy side to merge this..please take a look.

File: pkg/proxy/accesslog/log.go
Comment (raybejjani) [inline pkg/proxy/accesslog/log.go]: aanm has shown me a nifty trick. It reads nicer but feels completely weird. We can split between the chained functions, if you prefer:
```golang
log.WithField(FieldFilePath, logPath).
    Debug("Skipping writing to access log (write buffer nil)")
```

File: pkg/proxy/oxyproxy.go
Comment (raybejjani) [inline pkg/proxy/oxyproxy.go]: This is missing the Debug call
```golang
log.WithField(logfields.Object, fmt.Sprintf("%+v", ar).Debug("Allowing request based on rule")
```
We can also backport the logfields.Repr function but it basically just does the fmt.Sprintf call

File: pkg/policy/api/selector.go
Comment (raybejjani) [inline pkg/policy/api/selector.go]: This is missing the `Error` call
```golang
log.WithError(err).WithField(logfields.EndpointLabelSelector, fmt.Sprintf("%+v",n).
    Error("unable to match label selector in selector")
```

File: pkg/proxy/kafka.go
Comment (raybejjani) [inline pkg/proxy/kafka.go]: It might be more clear if we split the line before the func

**Review by raybejjani — CHANGES_REQUESTED** [review]
My comments are mostly about formatting or non-printing log messages. Feel free to override this if you prefer to fix those in a separate PR.

File: pkg/policy/api/selector.go
Comment (tgraf) [reply pkg/policy/api/selector.go]: fixed

File: pkg/proxy/accesslog/log.go
Comment (tgraf) [reply pkg/proxy/accesslog/log.go]: fixed

File: pkg/proxy/oxyproxy.go
Comment (tgraf) [reply pkg/proxy/oxyproxy.go]: fixed

File: pkg/proxy/kafka.go
Comment (tgraf) [reply pkg/proxy/kafka.go]: fixed

**Review by aanm — APPROVED** [review]

**Review by raybejjani — APPROVED** [review]

**Pull request merged**
